src/main/java/org/opensearch/security/auth/BackendRegistry.java

```diff
@@ -415,6 +404,22 @@ public boolean authenticate(final SecurityRequestChannel request) {
                 }
             }
 
+            if (authCredentials == null
+                && anonymousAuthEnabled
+                && isRequestForAnonymousLogin(request.params())
+                && !checkIfRequestContainsBasicAuthHeader(request.getHeaders())) {
```

Comment:
nit: This can be changed to `checkIfRequestContainsAuthHeader`

For proxy requests, JWT and client cert they won't send the credentials through the Authorization header so this is not a universal check to see if the request does not contain credentials

Reply:
Since the check is only for basic auth, i named it as such. For other types of auth this method should not be called unless `isRequestForAnonymousLogin(request.params())` succeeds.

Reply:
If the request is explicitly for anonymous login (i.e. Has a url param called `auth_request_type` with a value of `anonymous`), then do you think its necessary to check if the request has credentials or is credential-less?

In the scenario where anonymous auth is explicitly requested, should that ignore if credentials were also provided on the request?

Reply:
Anonymous requests will not contain a basic auth header, and  hence this check will return false which in the if condition check will be negated to true.

Reply:
I see where the confusion originated from. `isRequestForAnonymousLogin` returns true if no `auth_request_type` param was passed. To address this I've updated the companion front-end PR to avoid confusion here.